### PR TITLE
Refactor depthwise_conv to share code between reference and optimized kernels

### PR DESCRIPTION
**Pull request by njeffrie**
Move shared structs / helper functions into depthwise_conv_common.cc
Clean up some of the existing code to directly call the reference implementations (made possible by the refactor of the helper functions).

Tested with:
make -j8 -f tensorflow/lite/micro/tools/make/Makefile test_kernel_depthwise_conv_test

make -j8 -f tensorflow/lite/micro/tools/make/Makefile OPTIMIZED_KERNEL_DIR=cmsis_nn TARGET=stm32f4 test_kernel_depthwise_conv_test

make -f tensorflow/lite/micro/tools/make/Makefile -j8 TARGET=xtensa OPTIMIZED_KERNEL_DIR=xtensa TARGET_ARCH=hifimini XTENSA_CORE=mini1m1m_RG test_kernel_depthwise_conv_test

Progress towards http://b/177457688

### CODE REVIEW

**Review by advaitjain — APPROVED** [review]
Can you add a link to a bug from your PR description?

File: tensorflow/lite/micro/kernels/cmsis_nn/depthwise_conv.cc [hunk 1]
```diff
@@ -13,7 +13,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-#include "tensorflow/lite/kernels/internal/reference/integer_ops/depthwise_conv.h"
+#include "tensorflow/lite/micro/kernels/depthwise_conv.h"
 
 #include "CMSIS/NN/Include/arm_nnfunctions.h"
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/cmsis_nn/depthwise_conv.cc:18]: there should be a

```cc
#include "tensorflow/lite/micro/kernels/conv.h"
```

Right?

File: tensorflow/lite/micro/kernels/BUILD [hunk 2]
```diff
@@ -90,6 +90,43 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "depthwise_conv",
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/BUILD:94]: this should depend on :conv for the conv.h, right?

**Review by advaitjain — COMMENTED** [review]
small nits

File: tensorflow/lite/micro/kernels/BUILD (hunk 2 above)
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/BUILD:94]: It does below in the deps section, fixed by my second commit

File: tensorflow/lite/micro/kernels/cmsis_nn/depthwise_conv.cc (hunk 1 above)
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/cmsis_nn/depthwise_conv.cc:18]: Done.

**Review by advaitjain — APPROVED** [review]

**Comment by gbaned** [comment]
Seems auto-merge is not happening but the changes are merged into master now, so we can close this. Thank you for the PR.

**Pull request closed without merging**